src/GameServer/Bot/Population/ColdSimulationKernel.js

```
@@ -68,11 +72,157 @@ function deterministicRandom(state = {}) {
+function hasFiniteCoordinate(value) {
+    return value !== null
+        && value !== undefined
+        && String(value).trim() !== ''
+        && Number.isFinite(Number(value));
+}
+
+function routeDestination(state = {}, route = {}) {
+    return route?.destinations?.[String(state.characterId)] || route?.to || null;
+}
+
+function beginRouteTravelState(state = {}, route = null, timestamp = Date.now(), options = {}) {
```

ESLint

[error] 'options' is assigned a value but never used.

(no-unused-vars)